### PR TITLE
Updates to Blazor docs from 3.0

### LINKED ISSUE #13787: Review Blazor Sample code for API review changes
We've made a bunch of API changes in Blazor during preview 8 and preview 9. We need to review all of the existing docs and sample and find places where they are out of date.

### PR DESCRIPTION
**Pull request by rynowak**
[EDIT by guardrex to crosslink issue]

Fixes #13787

@guardrex 

I did a full pass through **all** of the docs and samples. Yup. **all**. 

I didn't find too many things that needed updates. I also added a few things where it seemed like we were light on explanation.

### CODE REVIEW

File: aspnetcore/blazor/components.md [hunk 1]
```diff
@@ -22,14 +22,6 @@ Components are implemented in [Razor](xref:mvc/views/razor) component files (*.r
 
 A component's name must start with an uppercase character. For example, *MyCoolComponent.razor* is valid, and *myCoolComponent.razor* is invalid.
 
-Components can be authored using the *.cshtml* file extension as long as the files are identified as Razor component files using the `_RazorComponentInclude` MSBuild property. For example, an app that specifies that all *.cshtml* files under the *Pages* folder should be treated as Razor components files:
-
-```xml
-<PropertyGroup>
-  <_RazorComponentInclude>Pages\**\*.cshtml</_RazorComponentInclude>
-</PropertyGroup>
```
Comment (rynowak) [inline aspnetcore/blazor/components.md]: This is an old workaround. We don't need to tell people about this anymore, I can't think of a case where we'd recommend it, and we might remove this in the future.

File: aspnetcore/blazor/forms-validation.md [hunk 2]
```diff
@@ -14,9 +14,6 @@ By [Daniel Roth](https://github.com/danroth27) and [Luke Latham](https://github.
 
 Forms and validation are supported in Blazor using [data annotations](xref:mvc/models/validation).
 
-> [!NOTE]
-> Forms and validation scenarios are likely to change with each preview release.
```
Comment (rynowak) [inline aspnetcore/blazor/forms-validation.md]: NOT ANYMORE. Now it's going on your **permanent record**

File: aspnetcore/blazor/state-management.md [hunk 3]
```diff
@@ -240,38 +240,20 @@ Prerendering might be useful for other pages that don't use `localStorage` or `s
...
 
 ... rendering code goes here ...
 
 @code {
     private int? currentCount;
-    private bool isWaitingForConnection;
-
-    protected override async Task OnInitializedAsync()
-    {
-        if (ComponentContext.IsConnected)
```
Comment (rynowak) [inline aspnetcore/blazor/state-management.md]: This hurts. I didn't realize we went to this level of optimization in the docs. We need to bring back a way to detect prerendering in 3.1.

@SteveSandersonMS @danroth27

File: aspnetcore/includes/blazor-prerendering.md [hunk 4]
```diff
@@ -60,31 +59,4 @@ Where `JSRuntime.InvokeAsync` is called, `ElementRef` is only used in `OnAfterRe
...
-<p>
-    Clicks:
-    <strong id="count">@count</strong>
-    <button id="increment-count" @onclick="@(() => count++)">Click me</button>
-</p>
-
-@code {
-    private int count;
-}
-```
```
Comment (rynowak) [inline aspnetcore/includes/blazor-prerendering.md]: RIP

**Comment by guardrex** [comment]
> I didn't find too many things that needed updates.

🍾⭐️🍕😁🎉🎈🎵👍☀️🚀🍻💃✨🎸🕺💕🥇🌹✈️:shipit:

... I think that response about sums up my enjoyment of reading that remark! *WOOT!*

I'll review and get this merged in the morning.

**Comment by rynowak** [comment]
Thanks!

**Pull request merged**